demostore_automation/tests/frontend/my_account/test_register_new_user.py

```diff
@@ -41,4 +42,79 @@ def test_register_valid_new_user(self):
         myacct.click_register_button()
 
         # verify user is registered
-        myacct_sin.verify_user_is_signed_in()
\ No newline at end of file
+        myacct_sin.verify_user_is_signed_in()
+
+        # logout
+        myacct_sin.logout_user()
```

Comment:
I personaly would not put this here. Because extra step that could fail my test.
You definitely should have a separate test to verify logout works. but I don't see the point of having a logout step here. But just an opinion ofcourse.

Reply:
I definitely see your point here. I added this logout function so that all of the tests in the file would run. After the test to register a user, the browser navigates to the "my account logged in" page, which results in a time-out error while the next test tries to look for the 'my account logged out' page. What is the best way to handle this?

Reply:
O I see how that is a problem.
Well the way you handled it is fine. But also for the next text you can have a 'setup' that verifies the user is logged out. So before the test starts the setup will make sure you are on the correct page. Checks if user is logged in then logg out or if user is not logged in just check if it is on the right page. Then the test starts.